src/main/java/org/codeforamerica/messaging/validators/MessageContentsValidator.java

```diff
@@ -0,0 +1,27 @@
+package org.codeforamerica.messaging.validators;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.apache.commons.lang3.StringUtils;
+import org.codeforamerica.messaging.models.Message;
+import org.codeforamerica.messaging.models.TemplateVariant;
+import org.codeforamerica.messaging.models.TemplateVariantRequest;
+
+public class MessageContentsValidator implements ConstraintValidator<ValidMessageContents, Object> {
+    @Override
+    public boolean isValid(Object object, ConstraintValidatorContext constraintValidatorContext) {
```

Comment:
Please consider having the objects implement a common interface, say `Messageable`, so that we can eliminate the `if-else`s?